base_layer/core/tests/tests/node_comms_interface.rs

```diff
@@ -251,6 +267,165 @@ async fn inbound_fetch_blocks() {
+            change.script_key_id,
+            change.spend_key_id,
+            Covenant::default(),
+        );
+
+    for tx_input in &txn_schema.from {
+        stx_builder.with_input(tx_input.clone()).await.unwrap();
+    }
+    for tx_output in txn_schema.to {
+        let (spending_key, _) = key_manager
+            .get_next_key(TransactionKeyManagerBranch::CommitmentMask.get_branch_key())
+            .await
+            .unwrap();
+        let (sender_offset_key_id, sender_offset_public_key) = key_manager
+            .get_next_key(TransactionKeyManagerBranch::SenderOffset.get_branch_key())
+            .await
+            .unwrap();
+        let (script_key_id, _) = key_manager
+            .get_next_key(TransactionKeyManagerBranch::ScriptKey.get_branch_key())
+            .await
+            .unwrap();
+        let script_public_key = key_manager.get_public_key_at_key_id(&script_key_id).await.unwrap();
+        let input_data = match &txn_schema.input_data {
+            Some(data) => data.clone(),
+            None => inputs!(script_public_key),
+        };
+        let version = match txn_schema.output_version {
+            Some(data) => data,
+            None => TransactionOutputVersion::get_current_version(),
+        };
+        let output = WalletOutputBuilder::new(tx_output, spending_key)
+            .with_features(txn_schema.features.clone())
+            .with_script(txn_schema.script.clone())
+            .encrypt_data_for_recovery(key_manager, None)
+            .await
+            .unwrap()
+            .with_input_data(input_data)
+            .with_covenant(txn_schema.covenant.clone())
+            .with_version(version)
+            .with_sender_offset_public_key(sender_offset_public_key)
+            .with_script_key(script_key_id.clone())
+            .sign_as_sender_and_receiver(key_manager, &sender_offset_key_id)
+            .await
+            .unwrap()
+            .try_build(key_manager)
+            .await
+            .unwrap();
+
+        stx_builder.with_output(output, sender_offset_key_id).await.unwrap();
+    }
+    for mut utxo in txn_schema.to_outputs {
+        let (sender_offset_key_id, _) = key_manager
+            .get_next_key(TransactionKeyManagerBranch::SenderOffset.get_branch_key())
+            .await
+            .unwrap();
+        let metadata_message = TransactionOutput::metadata_signature_message(&utxo);
+        utxo.metadata_signature = key_manager
+            .get_metadata_signature(
+                &utxo.spending_key_id,
+                &utxo.value.into(),
+                &sender_offset_key_id,
+                &utxo.version,
+                &metadata_message,
+                utxo.features.range_proof_type,
+            )
+            .await
+            .unwrap();
+
+        stx_builder.with_output(utxo, sender_offset_key_id).await.unwrap();
+    }
+
+    stx_builder
+}
+
+#[tokio::test]
+async fn test_sender_transaction_protocol_for_overflow() {
```

Comment:
This test passes even with the fee `saturating_add` reverted. Is this the intended behavior? I assumed this test was intended to catch a regression, which it apparently does not.

Reply:
I changed `pub fn add_kernels(mut self, mut kernels: Vec<TransactionKernel>)` without adding a regression test for it; this test is for the other three changes, will try to add test for `add_kernels`.